Lib/test/test_argparse.py

```diff
@@ -3114,12 +3114,6 @@ def test_set_defaults_on_parent_and_subparser(self):
         xparser.set_defaults(foo=2)
         self.assertEqual(NS(foo=2), parser.parse_args(['X']))
 
-    def test_set_defaults_on_subparser_with_namespace(self):
-        parser = argparse.ArgumentParser()
-        xparser = parser.add_subparsers().add_parser('X')
-        xparser.set_defaults(foo=1)
-        self.assertEqual(NS(foo=2), parser.parse_args(['X'], NS(foo=2)))
-
```

Comment:
How about we keep this test and mark is as `xfail` until we bring back a better patch for the original issue?

Reply:
I think it would be better to leave it out.  While there is uncertainty about what the correct behavior should be, there is certainty that this tested behavior is wrong.  Also, any future changes will likely only go onto the main branch, so there is no value in leaving disabled tests in older versions.

The open BPO issue will serve as the marker for future work to be done.